### DISCUSSION TITLE
Why minio takes up too much memory

### DISCUSSION THREAD

**Discussion by zhtttyl** [minio/minio#11622] [Q&A]
Hey:
When testing WordCount 1T data, Minio used a lot of memory in the reduce phase.The number of reduce is 2048
![minioPng](https://github.com/zhtttyl/miniotest/blob/main/minio13G.png?raw=true)

I run  `mc admin profile start --type=cpu,mem myminio`
https://raw.githubusercontent.com/zhtttyl/miniotest/main/profile.zip

**Comment by klauspost** [comment]
Lots of concurrent writes going on. Mainly `PutObjectPart` calls.

If you need to limit concurrency use `mc admin config set myminio api requests_max=n` to limit the number of concurrent operations.

**Comment by harshavardhana** [comment]
Surprisingly you are asking why is MinIO to using more memory while at the same time you are running a Java application on same node eating up more memory than MinIO.
